test/algos/adaptors/test_split.cpp

```diff
@@ -19,26 +20,185 @@
+  ex::start(op);
+  // The receiver will ensure that the right value is produced
+}
+TEST_CASE("split forwards results from a different thread", "[adaptors][split]") {
+  example::static_thread_pool pool{1};
+  auto split = ex::schedule(pool.get_scheduler()) | //
+               ex::then([] {
+                 std::this_thread::sleep_for(1ms);
+                 return 42;
+               }) | //
+               ex::split();
+
+  auto [val] = std::this_thread::sync_wait(split).value();
+  REQUIRE( val == 42 );
+}
+TEST_CASE("split is thread-safe", "[adaptors][split]") {
+  example::static_thread_pool pool{1};
+
+  std::mt19937_64 eng{std::random_device{}()};  // or seed however you want
+  std::uniform_int_distribution<> dist{0, 1000};
+
+  auto split = ex::transfer_just(pool.get_scheduler(), std::chrono::microseconds{dist(eng)}) | //
+               ex::then([] (std::chrono::microseconds delay) {
+                 std::this_thread::sleep_for(delay);
+                 return 42;
+               }) | //
+               ex::split();
+
+  const unsigned n_threads = std::thread::hardware_concurrency();
+  std::vector<std::thread> threads(n_threads);
+  const std::vector<std::chrono::microseconds> delays = [&] {
+    std::vector<std::chrono::microseconds> thread_delay(n_threads);
+    for (unsigned tid = 0; tid < n_threads; tid++) {
+      thread_delay[tid] = std::chrono::microseconds{dist(eng)};
+    }
+    return thread_delay;
+  }();
+  for (unsigned tid = 0; tid < n_threads; tid++) {
+    threads[tid] = std::thread([&split, &delays, tid] {
+      example::inline_scheduler scheduler{};
+
+      std::this_thread::sleep_for(delays[tid]);
+      auto [val] = std::this_thread::sync_wait(
+          split |                   //
+          ex::transfer(scheduler) | //
+          ex::then([](int v) { return v; })).value();
+      REQUIRE( val == 42 );
+    });
+  }
+  for (auto& thread: threads) {
+    thread.join();
+  }
+}
+TEST_CASE("split can be an rvalue", "[adaptors][split]") {
+  auto [val] = std::this_thread::sync_wait(
+      ex::just(42) |
+      ex::split() |
+      ex::then([](int v) { return v; } )).value();
+
+  REQUIRE( val == 42 );
+}
+TEST_CASE("split can nest", "[adaptors][split]") {
+  auto split_1 = ex::just(42) | ex::split();
+  auto split_2 = split_1 | ex::split();
+
+  auto [v1] = std::this_thread::sync_wait(
+      split_1 | //
+      ex::then([](int &v) { return v = 1; })).value();
+
+  auto [v2] = std::this_thread::sync_wait(
+      split_2 | //
+      ex::then([](int &v) { return v = 2; })).value();
+
+  auto [v3] = std::this_thread::sync_wait(split_1).value();
+
+  REQUIRE( v1 == 1 );
+  REQUIRE( v2 == 2 );
+  REQUIRE( v3 == 1 );
+}
+TEST_CASE("split advertises completion scheduler", "[adaptors][split]") {
+  inline_scheduler sched;
+
+  SECTION("for value channel")
+  {
+    auto snd = ex::transfer_just(sched, 42) | ex::split();
+    REQUIRE(ex::get_completion_scheduler<ex::set_value_t>(snd) == sched);
+  }
+}
```

Comment:
Would be nice to have cancellation tests also. Bit tricky that. I would be ok with filing an issue and doing it later.